librarian: detect formatting may fail due to a timing diff

Observed failure related to `cmd/librarian/doc.go` although the change did not touch anything related. See log: https://github.com/googleapis/librarian/actions/runs/19443688708/job/55632944683?pr=2931

This error went away with a re-run. This is a rare condition that I have not seen happen before.

The error `Error: +./doc.go:1:1: syntax error: package statement must be first` suggests that the file `cmd/librarian/doc.go` was in a corrupted or incomplete state at the moment `git diff` tried to read it.

It's likely, `go generate` (triggered by the test in the previous step) was still in the process of writing to cmd/librarian/doc.go when the "Run tests" step finished and the "Detect formatting changes" step immediately started. According to gemini, while workflow steps are sequential, there can be a slight delay between a process exiting and its file system changes being fully flushed and stable.

--- a/.github/workflows/librarian.yaml
+++ b/.github/workflows/librarian.yaml
@@ -36,8 +36,6 @@ jobs:
           go install golang.org/x/tools/cmd/goimports@latest
       - name: Run tests
         run: go test -race -coverprofile=coverage.out -covermode=atomic ./...
-      - name: Detect formatting changes
-        run: git diff --exit-code
       - name: Upload coverage report
         uses: codecov/codecov-action@v5
         with:

